server/src/controllers/grades.ts

```diff
@@ -758,8 +758,21 @@ export function parseGradesFromCsv(
  * the CSV file contains attainments which don't belong to the specified course or course instance.
  */
 export async function addGrades(req: Request, res: Response, next: NextFunction): Promise<void> {
-  /** TODO: Check grading points are not higher than max points of the attainment. */
+  const requestSchema: yup.AnyObjectSchema = yup.object().shape({
+    completionDate: yup
+      .date()
+      .notRequired(),
+    expiryDate: yup
+      .date()
+      .notRequired()
```

Comment:
It should be checked that this is greater than or equal to `completionDate`. Now that I think about it, that would be a good constraint to add too. Does Postgres allow for a constraint of the form `Date | null >= Date` (see #497) or `Date | null >= Date | null` (as it is now)?

Reply:
Added both yup and database checks. Postgres is okay even if there are null values, these are basically just skipped